ivre/db/sql/__init__.py

```diff
@@ -809,13 +811,82 @@ def __migrate_schema_9_10(self):
         )
         return len(failed)
 
-    def __migrate_schema_10_11(self):
+    def _migrate_schema_10_11(self):
         """Converts a record from version 10 to version 11. Version 11 changes
 the way IP addresses are stored.
 
         """
         raise NotImplementedError
 
+    def _migrate_schema_11_12(self):
+        """Converts a record from version 11 to version 12. Version 12 changes
+the structured output for fcrdns and rpcinfo script.
+
+        """
+        failed = set()
+        req = (select([self.tables.scan.id,
+                       self.tables.script.port,
+                       self.tables.script.output,
+                       self.tables.script.data])
+               .select_from(join(join(self.tables.scan, self.tables.port),
+                                 self.tables.script))
+               .where(and_(self.tables.scan.schema_version == 11,
+                           self.tables.script.name == "fcrdns")))
+        for rec in self.db.execute(req):
+            if 'fcrdns' in rec.data:
+                try:
+                    # Since dots in keys are converted to
+                    # underscores, we need to convert them
+                    # back. It's OK to do so because underscores
+                    # cannot be used in FQDNs.
+                    data = [
```

Comment:
You might want to use your function `change_fcrdns`.

Reply:
Same.